supabase/utils/selectLoggedUserId.ts

```diff
@@ -0,0 +1,16 @@
+'use server';
+
+import newServerClient from './newServerClient';
+
+export default async function selectLoggedUserId() {
+  const supabase = newServerClient();
+  try {
+    const { data, error } = await supabase.auth.getUser();
+
+    if (error) throw error;
+
+    return data.user.id;
+  } catch (error) {
+    console.error('Error getting user session: ', error);
```

Comment:
throw your error here again after logging or it won't be caught in the client side when we call this function